src/loaders/ncbi_downloader/ncbi_downloader.py

```diff
@@ -34,18 +110,92 @@ def _download_genome_file(download_dir: str, gene_id: str, target_file_ext: list
                 for gene_file_name in gene_file_list:
                     if any([gene_file_name.endswith(ext) for ext in target_file_ext]):
                         result_file_path = os.path.join(download_dir, gene_file_name)
-                        host.download(gene_file_name, result_file_path)
+                        if overwrite or not os.path.exists(result_file_path):
+                            host.download(gene_file_name, result_file_path)
 
             success = True
         except Exception as e:
-            print(e)
+            print(f'Error:\n{e}\nfrom attempt {attempts + 1}.\nTrying to rerun.')
             attempts += 1
 
     if not success:
         raise ValueError(f'Download Failed for {gene_id} after {max_attempts} attempts!')
 
 
-def download_genome_files(gene_ids: list[str], target_file_ext: list[str], result_dir='ncbi_genomes') -> list[str]:
+def _form_gtdb_taxonomy_file_url(release_ver):
+    # form GTDB taxonomy URL for specific GTDB version.
+    # e.g. https://data.gtdb.ecogenomic.org/releases/release207/207.0/ar53_taxonomy_r207.tsv
+
+    if release_ver not in GTDB_RELEASE_VER:
+        raise ValueError(f'Unsupported GTDB release version: {release_ver}')
+
+    file_dir_url = GTDB_DOMAIN + f'release{release_ver}/' + f'{release_ver}.0/'
+
+    file_urls = list()
+
+    # parse taxonomy file URL for genome ids
+    response = requests.get(file_dir_url)
+    soup = BeautifulSoup(response.text, 'html.parser')
+
+    for link in soup.find_all('a'):
+        href = link.get('href')
+
+        if href.endswith('tsv') and '_taxonomy_' in href:
+            file_urls.append(file_dir_url + href)
+
+    if not file_urls or len(file_urls) >= 3:
+        raise ValueError(
+            f'Parsed unexpected taxonomy files {file_urls}. '
+            f'Please check parsing logic in _form_gtdb_taxonomy_file_url.')
+
+    return file_urls
+
+
+def _fetch_gtdb_genome_ids(release_ver, work_dir):
+    # download GTDB taxonomy files and then parse genome_ids from the GTDB taxonomy files
+    genome_ids = list()
+
+    taxonomy_urls = _form_gtdb_taxonomy_file_url(release_ver)
+
+    for taxonomy_url in taxonomy_urls:
+        # download GTDB taxonomy file to work_dir
+        url = urlparse(taxonomy_url)
+        taxonomy_file = os.path.join(work_dir, os.path.basename(url.path))
+        _ = request.urlretrieve(taxonomy_url, taxonomy_file)
```

Comment:
Is there a reason to have the `=` at all?

Reply:
nope. removed.

Reply:
:+1: